test/document_stores/test_weaviate.py

```diff
@@ -144,3 +144,46 @@ def test_deleting_by_id_or_by_filters(document_store_with_docs):
     document_store_with_docs.delete_documents(filters={"name": ["filename2"]})
     document_store_with_docs.get_all_documents.assert_called()
     assert document_store_with_docs.get_document_count() == 1
+
+
+def get_document_store_with_docs_and_similarity(tmp_path, similarity, recreate_index):
+    document_store = get_document_store(
+        "weaviate", tmp_path=tmp_path, similarity=similarity, recreate_index=recreate_index
+    )
+    document_store.write_documents(DOCUMENTS_XS)
+    yield document_store
+    document_store.delete_index(document_store.index)
+
+
+@pytest.mark.weaviate
+@pytest.mark.parametrize("similarity", ["cosine", "l2", "dot_product"])
+def test_similarity(tmp_path, similarity):
+    document_store = next(get_document_store_with_docs_and_similarity(tmp_path, similarity, recreate_index=True))
+    docs = document_store.query_by_embedding(np.random.rand(embedding_dim).astype(np.float32))
+
+    assert len(docs) == 3
+
+    if similarity == "cosine":
+        assert document_store.similarity == "cosine"
+    elif similarity == "dot_product":
+        assert document_store.similarity == "dot"
+    elif similarity == "l2":
+        assert document_store.similarity == "l2-squared"
+
+    for doc in docs:
+        assert 0 <= doc.score <= 1
+
+
+@pytest.mark.weaviate
+@pytest.mark.parametrize("similarity", ["cosine", "l2", "dot_product"])
+def test_similarity_existing_index(tmp_path, similarity):
+    """Testing non-matching similarity"""
+    # create the document_store
+    document_store = next(get_document_store_with_docs_and_similarity(tmp_path, similarity, recreate_index=True))
```

Comment:
I think you can substitute `get_document_store_with_docs_and_similarity` by `get_document_store` from `conftest.py`.

Reply:
You are right, I have substituted the `get_document_store_with_docs_and_similarity` with the `get_document_store` from `conftest.py`.